feat(Grid): Allow cells to span multiple rows (#2537)
Amsterdam/design-system: storybook/src/components/Grid/Grid.docs.mdx

<Canvas of={GridStories.SpanColumns} />

### Span rows

The `rowSpan` prop makes a Cell span multiple rows.

<Canvas of={GridStories.SpanRows} />

### Span responsively

> You can make the number of columns a cell spans depend on the window width.

Author: ```suggestion
The number of columns or rows that a cell covers can vary with the window width.
```

Use the `span` or `rowSpan` prop with 3 values for narrow, medium, and wide windows.

> E.g. `span={{ narrow: 4, medium: 5, wide: 7 }}` makes the cell span all 4 columns on narrow windows, 5 of the 8 on medium windows, and 7 of the 12 on wide windows.

Author: Updated.
